crates/typst/src/introspection/query.rs

```diff
@@ -13,58 +13,53 @@ use crate::introspection::Location;
 ///
 
 /// # Finding elements
-/// In the example below, we create a custom page header that displays the text
-/// "Typst Academy" in small capitals and the current section title. On the
-/// first page, the section title is omitted because the header is before the
-/// first section heading.
+/// In the example below, we manually create a table of contents instead of
+/// using the [`outline`] function.
 ///
-/// To realize this layout, we open a `context` and then query for all headings
-/// after the [current location]($here). The code within the context block
-/// runs twice: Once per page.
+/// To do this, we first query for all headings in the document at level 1 and
+/// after the current location. Querying only for headings at level 1 ensures
+/// that sub-headings are not included in the table of contents and after
+/// `here()` ensures that the "Table of Contents" heading is not included.
```

Comment:
I think a more idiomatic approach would be to filter for `heading.outlined` and use `heading(outlined: false)[Table of Contents]`. Alternatively, the example could specifically be about a special kind of outline, that only outlines the headings before/after it. That way, we can demo before/after + `here()`.